nltk/sem/relextract.py

```diff
@@ -198,17 +198,20 @@ def extract_rels(subjclass, objclass, doc, corpus='ace', pattern=None, window=10
     :return: see ``mk_reldicts``
     :rtype: list(defaultdict)
     """
-
-    if subjclass and subjclass not in NE_CLASSES[corpus]:
-        if _expand(subjclass) in NE_CLASSES[corpus]:
-            subjclass = _expand(subjclass)
-        else:
-            raise ValueError("your value for the subject type has not been recognized: %s" % subjclass)
-    if objclass and objclass not in NE_CLASSES[corpus]:
-        if _expand(objclass) in NE_CLASSES[corpus]:
-            objclass = _expand(objclass)
-        else:
-            raise ValueError("your value for the object type has not been recognized: %s" % objclass)
+    
+    for i, s in enumerate(subjclass):
+        if s and s not in NE_CLASSES[corpus]:
+            if _expand(s) in NE_CLASSES[corpus]:
+                subjclass[i] = _expand(s)
+            else:
+                raise ValueError("your value for the subject type has not been recognized: %s" % s)
+
+    for i, o in enumerate(objclass):
```

Comment:
@dimazest it could be moved to a helper function but I'm not sure any other piece of code would use that helper function.  Sorry to ask again but can you clarify why you think this needs an exception?  If the class is in the corpus list of accepted classes then it passes through so we don't need an extra exception right?  The point of these checks are to see if we need to expand the selected class or not.  Maybe I'm not understanding your question here.